src/Security/Security.php

```diff
@@ -336,59 +444,14 @@ public static function permissionFailure($controller = null, $messageSet = null)
         ));
     }
 
-    protected function init()
-    {
-        parent::init();
-
-        // Prevent clickjacking, see https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options
-        $frameOptions = $this->config()->get('frame_options');
-        if ($frameOptions) {
-            $this->getResponse()->addHeader('X-Frame-Options', $frameOptions);
-        }
-
-        // Prevent search engines from indexing the login page
-        $robotsTag = $this->config()->get('robots_tag');
-        if ($robotsTag) {
-            $this->getResponse()->addHeader('X-Robots-Tag', $robotsTag);
-        }
-    }
-
-    public function index()
-    {
-        return $this->httpError(404); // no-op
-    }
-
-    /**
-     * Get the selected authenticator for this request
-     *
-     * @return string Class name of Authenticator
-     * @throws LogicException
-     */
-    protected function getAuthenticator()
+    public static function setCurrentUser($currentUser = null)
```

Comment:
I suggest to type-cast the argument as a nullable Member instance.
public static function setCurrentUser(Member $currentUser = null)